lsl_autobids/processing_new_files.py

```diff
@@ -1,154 +1,123 @@
-from globals import project_root
+from config_globals import cli_args, project_root
 
-def proceesing_new_files(file_status, project_name):
-    """
-    Processes the new files with the ".xdf" extension, adding the file names (without "_old") to a list.
-    
-    Parameters
-    ----------
-    file_status : list
-        A list of new files found since the last run
-    project_name : str
-        The name of the project
-
-    Returns
-    -------
-    None
-        
+
+# Set up logging
+logger = logging.getLogger(__name__)
+logging.basicConfig(level=logging.INFO, format='%(levelname)s: %(message)s')
+
+
+def process_new_files(file_status: List[str]) -> None:
+    """Processes new .xdf files and prompts user to convert/upload.
+
+    Args:
+        file_status (List[str]): List of new files detected.
     """
-    print("Processing new files.........")
+
+    logger.info("Processing new files...")
+    # set up global variables
+    project_name = cli_args.project_name
     processed_files = []
     for file_path in file_status:
         if file_path.endswith('.xdf'):
             
             # This code is to handle the recordings from the same session and same subject
             # Remove "_old" from the filename if present
             _, file_name = os.path.split(file_path)
-            file_name_without_ext, ext = os.path.splitext(file_name)
-            if file_name_without_ext.endswith('_old'):
-                warning_message = "WARNING: The file already exists. It might be an old file. Are you sure you want to overwrite it? (y/n): "
-                while True:
-                    user_response = input(warning_message)
-                    if user_response.lower() == 'n':
-                        warnings.warn("Operation aborted. File will not be overwritten. Please check the file again!", UserWarning)
-                        sys.exit()
-                    elif user_response.lower() == 'y':
-                        print('Operation resumed. File will be overwritten with the new file....')
-                        break
-                    else:
-                        print('Invalid response. Please enter "y" for yes or "n" for no.')
-
-                # Skip adding the file with the _old to processed_files if the user chooses to overwrite
-                if user_response.lower() == 'y':
-                    continue  
-            processed_files.append(file_name_without_ext + ext)
-    print("The processed files are: \n")
-    print(processed_files)
+            file_name_no_ext, ext = os.path.splitext(file_name)
+            if file_name_no_ext.endswith('_old'):
+                user_choice = get_user_input("WARNING: The file appears to be a duplicate ('_old'). Overwrite?")
+                if user_choice == 'n':
+                    logger.warning("User declined to overwrite file. Exiting.")
+                    raise RuntimeError("File overwrite declined by user.")
+                logger.info("Overwrite confirmed.")
```

Comment:
This is dangerous. If the user presses anything else than 'n' it will be overwritten. If anything, the behavior should be the opposite. In case of doubt, don't overwrite

Reply:
Also in this loop, only once user is asked and upon wrong input not asked again to correct..maybe we could have a function doing the user input for all places with  inputs?